include/sys/dmu_objset.h

```diff
@@ -71,6 +71,19 @@ typedef struct objset_phys {
 
 typedef int (*dmu_objset_upgrade_cb_t)(objset_t *);
 
+#define	OS_DNODE_LISTS	16
```

Comment:
It would be nice to make this tunable as a module option.  That would be make it easier to do some performance testing to determine the best default value for this.